src/query/functions/binary/binary.go

```diff
@@ -191,6 +191,10 @@ func processBothSeries(
 	lMeta.ResultMetadata = lMeta.ResultMetadata.
 		CombineMetadata(rMeta.ResultMetadata)
 	// Use metas from only taken left series
+
+	if matching.On && matching.Card == CardOneToOne && len(matching.MatchingLabels) > 0 {
+		lMeta.Tags = models.EmptyTags()
```

Comment:
Would it be possible to add a test for this change in `binary_test.go`?

Reply:
Doesn't this clear the tags on the result? I think it should be `lMeta.Tags = lMeta.Tags.TagsWithKeys(matching.Matching)` or something along those lines?

Reply:
It does clear only extra tags but actually, when I was writing a test I found one more case when not all required tags get cleared. So I moved that code block to `intersect` method which makes more sense and I have added compatibility test too.